### PR TITLE
chore: Sync Endo versions

### PR DESCRIPTION
**Pull request by mhofman**
closes: #4413 

## Description

Syncs the versions of endo packages to the latest published versions.
Removes outdated patch files for endo packages
Includes the changes from https://github.com/Agoric/agoric-sdk/pull/4413 (minus patch files) to bring in compatibility with explicit types for `@endo/marshal`
Remove extraneous eslint global declarations

### Testing Considerations

This change is depends on concurrent fixes which finish dapp migrations from -r esm to Node.js native ESM.

- https://github.com/Agoric/documentation/pull/622
- https://github.com/Agoric/dapp-otc/pull/31

### CODE REVIEW

File: packages/xsnap/src/xsnap.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /* global process */
-// @ts-check
+// @ts-nocheck
```
Comment (mhofman) [inline packages/xsnap/src/xsnap.js]: This is the change we need to revert

**Review by michaelfig — APPROVED** [review]
LGTM!

**Comment by mhofman** [comment]
I'd like to do a little yarn surgery tomorrow morning to avoid duplicate babel packages (which may be a reason dapp broke?)

**Comment by kriskowal** [comment]
> I'd like to do a little yarn surgery tomorrow morning to avoid duplicate babel packages (which may be a reason dapp broke?)

This almost certainly contributed, as it would explain why I had to explicitly pre-initialize `@agoric/babel-standalone` in a couple places. It doesn’t account for the problem with `@babel/types` not having a default export when run under `-r esm`, which is a dependency we brought in via `@endo/static-module-record` depending on `recast` depending on `@babel/types`.

**Pull request merged**